### PR TITLE
Change deployment to be based on use cases #9

### LINKED ISSUE #9: Examples should be simple and use case driven
### Orb version

0.3.0

### What happened

Found a bunch of commands that were unclear purpose

### Expected behavior

Expected to find a simple "update" or "deploy" orb example

### more
Current example https://circleci.com/orbs/registry/orb/circleci/kubernetes#usage-deployment
1) we shouldnt need minukube to deploy, thats a testing use case right?
2) The deploy example should not delete resources
3)we should be clear about the different intentions of apply, update, etc

### PR DESCRIPTION
**Pull request by slievr**
### Checklist

- [x] All new jobs, commands, executors, parameters have descriptions
- [x] Examples have been added for any significant new features
- [x] README has been updated, if necessary

### Motivation, issues

Breaking out deployment examples into use cases

Resolves #9 

### Description

added example for creating, updating, and deleting

### CODE REVIEW

File: src/examples/delete.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+description: |
+  Deploying kubernetes resources.
```
Comment (KyleTryon) [inline src/examples/delete.yml]: Is this description accurate? This seems to delete a k8 resource.

Comment (slievr) [reply src/examples/delete.yml]: You are completely correct, I'll update that

**Comment by lokst** [comment]
Thanks @slievr, we'll review this and get back to you as soon as possible

File: src/examples/delete.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  version: 2.1
+
+  orbs:
+    kube-orb: circleci/kubernetes@x.y.z
+
+  jobs:
+    build:
+      machine:
+        image: circleci/classic:201808-01
+      environment: CHANGE_MINIKUBE_NONE_USER=true
```
Comment (lokst) [inline src/examples/delete.yml]: Could you remove this since the example doesn't make use of minikube?

File: src/examples/delete.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+usage:
+  version: 2.1
+
+  orbs:
+    kube-orb: circleci/kubernetes@x.y.z
+
+  jobs:
+    build:
+      machine:
```
Comment (lokst) [inline src/examples/delete.yml]: As the example doesn't rely on minikube, I think there's no special reason to use the machine executor. Could we use the docker executor instead?

For example: 
```
docker:
      - image: cimg/base:stable
```

File: src/examples/deployment.yml [hunk 4]
```diff
@@ -16,25 +16,8 @@ usage:
       steps:
         - checkout
         - kube-orb/install-kubectl
-        - run:
```
Comment (lokst) [inline src/examples/deployment.yml]: Thanks for removing this! Could you also remove the `CHANGE_MINIKUBE_NONE_USER` environment variable and change the executor like the following?

```
docker:
      - image: cimg/base:stable
```

File: src/examples/deployment.yml [hunk 5]
```diff
@@ -16,25 +16,8 @@ usage:
...
-              curl -Lo minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64 \
-                && chmod +x minikube
-              sudo cp minikube /usr/local/bin && rm minikube
-              sudo -E minikube start --vm-driver=none --cpus 2 --memory 2048
         - kube-orb/create-or-update-resource:
             resource-file-path: "tests/nginx-deployment/deployment.yaml"
             resource-name: "deployment/nginx-deployment"
             get-rollout-status: true
             show-kubectl-command: true
-        - kube-orb/update-container-image:
```
Comment (lokst) [inline src/examples/deployment.yml]: Could you move the `kube-orb/update-container-image` step to `src/examples/deployment-update.yml` (which seems to be an empty file at the moment) ?

File: src/examples/delete.yml (hunk 3 above)
Comment (slievr) [reply src/examples/delete.yml]: I've fixed this in https://github.com/CircleCI-Public/kubernetes-orb/pull/26/commits/5902ad985f23ddd77834390600923a5d15e418c1 gave the job a more descriptive name too

File: src/examples/deployment.yml (hunk 5 above)
Comment (slievr) [reply src/examples/deployment.yml]: I thought I'd done this already, sorry. Updated in https://github.com/CircleCI-Public/kubernetes-orb/pull/26/commits/9fafd60156a64043daeee4bea83d351893fe3e0e

File: src/examples/deployment.yml (hunk 4 above)
Comment (slievr) [reply src/examples/deployment.yml]: fixed in https://github.com/CircleCI-Public/kubernetes-orb/pull/26/commits/02a7350ea8c3166c48082f5ae9a2b83d55aa3e47

**Comment by slievr** [comment]
@lokst I've gone through your comments and made changes as requested, let me know if there's anything else.

**Pull request merged**